version: '3'
services:
    webserver:
        image: raazpuspa/larasail:8.1
        container_name: '${APP_NAME:-reservoir}'
        extra_hosts:
            - 'host.docker.internal:host-gateway'
        ports:
            - '${APP_PORT:-4040}:80'
            - '${VITE_PORT:-4041}:${VITE_PORT:-4041}'
        environment:
            LARAVEL_SAIL: 1
            WWWUSER: '${WWWUSER:-1000}'
            WWWGROUP: '${WWWGROUP:-1000}'
            XDEBUG_MODE: '${SAIL_XDEBUG_MODE:-off}'
            XDEBUG_CONFIG: '${SAIL_XDEBUG_CONFIG:-client_host=host.docker.internal}'
        volumes:
            - '.:/var/www/html'
        networks:
            - internal
        depends_on:
            - database
    database:
        image: 'postgres:14'
        container_name: '${APP_NAME:-reservoir}_database'
        ports:
            - '${FORWARD_DB_PORT:-4042}:5432'
        environment:
            PGPASSWORD: '${DB_PASSWORD:-reservoir}'
            POSTGRES_DB: '${DB_DATABASE:-reservoir}'
            POSTGRES_USER: '${DB_USERNAME:-reservoir}'
            POSTGRES_PASSWORD: '${DB_PASSWORD:-reservoir}'
        volumes:
            - 'database:/var/lib/postgresql/data'
        networks:
            - internal
        healthcheck:
            test: [ "CMD", "pg_isready", "-q", "-d", "${DB_DATABASE:-reservoir}", "-U", "${DB_USERNAME:-reservoir}" ]
            retries: 3
            timeout: 5s
networks:
    internal:
        driver: bridge
volumes:
    database:
        driver: local
